# 1. Table 1

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import statsmodels.api as sm
import statsmodels.formula.api as smf

In [2]:
df = pd.read_csv("charitable_giving.csv")
df.head()

donation_amount  donation_dummy  control  treatment  match_ratio  ratio1  \
0              0.0             0.0      0.0        1.0          1.0       1   
1              0.0             0.0      1.0        0.0          0.0       0   
2              0.0             0.0      1.0        0.0          0.0       0   
3              0.0             0.0      0.0        1.0          3.0       0   
4              0.0             0.0      0.0        1.0          2.0       0   

   ratio2  ratio3  red_state_dummy  months_since_last_donation  \
0     0.0     0.0              1.0                        19.0   
1     0.0     0.0              1.0                        29.0   
2     0.0     0.0              1.0                         3.0   
3     0.0     1.0              0.0                         4.0   
4     1.0     0.0              0.0                         8.0   

   highest_previous_donation  prior_donations_num  
0                      500.0                 32.0  
1                      300.0                 22.0  
2                      500.0                 22.0  
3                      250.0                 29.0  
4                       50.0                 17.0

In [3]:
df.describe()

donation_amount  donation_dummy       control     treatment  \
count     50083.000000    50083.000000  50083.000000  50083.000000   
mean          0.915694        0.020646      0.333187      0.666813   
std           8.709199        0.142197      0.471357      0.471357   
min           0.000000        0.000000      0.000000      0.000000   
25%           0.000000        0.000000      0.000000      0.000000   
50%           0.000000        0.000000      0.000000      1.000000   
75%           0.000000        0.000000      1.000000      1.000000   
max         400.000000        1.000000      1.000000      1.000000   

        match_ratio        ratio1        ratio2        ratio3  \
count  50083.000000  50130.000000  50083.000000  50083.000000   
mean       1.333546      0.222083      0.222311      0.222211   
std        1.154606      0.415651      0.415803      0.415736   
min        0.000000      0.000000      0.000000      0.000000   
25%        0.000000      0.000000      0.000000      0.000000   
50%        1.000000      0.000000      0.000000      0.000000   
75%        2.000000      0.000000      0.000000      0.000000   
max        3.000000      1.000000      1.000000      1.000000   

       red_state_dummy  months_since_last_donation  highest_previous_donation  \
count     50048.000000                50082.000000               50083.000000   
mean          0.404452                   13.007268                  59.384977   
std           0.490791                   12.081403                  71.177310   
min           0.000000                    0.000000                   0.000000   
25%           0.000000                    4.000000                  30.000000   
50%           0.000000                    8.000000                  45.000000   
75%           1.000000                   19.000000                  60.000000   
max           1.000000                  168.000000                1000.000000   

       prior_donations_num  
count         50083.000000  
mean              8.039355  
std              11.394454  
min               0.000000  
25%               2.000000  
50%               4.000000  
75%              10.000000  
max             218.000000

In [4]:
reg = smf.ols(formula = 'months_since_last_donation ~ treatment', data = df)
result = reg.fit()
print(result.summary())

                                OLS Regression Results                                
Dep. Variable:     months_since_last_donation   R-squared:                       0.000
Model:                                    OLS   Adj. R-squared:                 -0.000
Method:                         Least Squares   F-statistic:                   0.01428
Date:                        Sat, 25 Jan 2025   Prob (F-statistic):              0.905
Time:                                10:45:56   Log-Likelihood:            -1.9585e+05
No. Observations:                       50082   AIC:                         3.917e+05
Df Residuals:                           50080   BIC:                         3.917e+05
Df Model:                                   1                                         
Covariance Type:                    nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------

**Question 1. Load the “charitable giving.csv” dataset and run a regression to assess whether the average “Number of months since last donation” is significantly different between treatment and control. Interpret the relevant regression coefficients and compare the regression-based comparison to the group-specific means reported in Table 1 of the paper.**

Ans:  
Intercept: 12.9981 (mean number of months since last donation for the control group)  
Treatment Coefficient: 0.0137 (indicates the treatment group's mean is higher than the control group by 0.0137 months)  
P value: 0.905 is bigger than 0.05 which means the difference is not statistically significant

Since treatment in the regression (12.9981 + 0.0137 = 13.0118) is very close to the treatment (13.012) in Table 1 so does control (12.998), we can know they have nearly identical means for "Number of months since last donation," and the difference (0.0137 & 0.014) is not significant.

**Question 2. Is the difference in “Number of month since last donation” between treatment and control statistically significant (at the usual 95% confidence level)? Is this the result you expected?**

Ans:  
Since p value is 0.905 far bigger than 0.05, the difference is not statistically significant. It is what we expected as we can see from the difference (13.012 - 12.998 = 0.014 months.)

**Question 3. More generally, describe the take-away from Table 1 in the paper.**

Ans:  
Table 1 proves that the treatment and control are balanced, approximatedly 13 months. Second, the statistically insignificance in difference means that the treatment doesn't significantly impact the number of months since last donation.

# 2. Response rate regressions

In [5]:
reg1 = smf.ols(formula = 'donation_dummy ~ treatment', data = df)
result1 = reg1.fit()
print(result1.summary())

                            OLS Regression Results                            
Dep. Variable:         donation_dummy   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                     9.618
Date:                Sat, 25 Jan 2025   Prob (F-statistic):            0.00193
Time:                        10:45:56   Log-Likelihood:                 26630.
No. Observations:               50083   AIC:                        -5.326e+04
Df Residuals:                   50081   BIC:                        -5.324e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.0179      0.001     16.225      0.0

**Question 1. Run a linear regression of response rate (the donation dummy) on the treatment dummy (and an intercept). Interpret both coefficients and compare them to the results presented in the first row of Table 2a.**

Ans:  
Intercept: 0.0179 (When treatment is 0, intercept represents the average response rate for control group)  
Treatment: 0.0042 (indicates the treatment group's average response rate is higher than the control group by 0.42%)     

Since treatment in the regression (0.0042 + 0.0179 = 0.0221) is very close to the treatment (0.022) in Table 2 so does control (0.018), we can know they have nearly identical "average response rate."


In [9]:
match_ratio1 = (df['match_ratio'] == 1)
match_ratio2 = (df['match_ratio'] == 2)
match_ratio3 = (df['match_ratio'] == 3)

reg2 = smf.ols(formula = 'donation_dummy ~ match_ratio1 + match_ratio2 + match_ratio3', data = df)
result2 = reg2.fit()
print(result2.summary())

                            OLS Regression Results                            
Dep. Variable:         donation_dummy   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                     3.665
Date:                Sat, 25 Jan 2025   Prob (F-statistic):             0.0118
Time:                        11:21:25   Log-Likelihood:                 26630.
No. Observations:               50083   AIC:                        -5.325e+04
Df Residuals:                   50079   BIC:                        -5.322e+04
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
Intercept                0.0179 

**Question 2. Run a regression on three dummies for match ratio treatment (1:1, 2:1, and 3:1 and an intercept). Interpret all four regression coefficients.**  

Ans:  
Intercept: 0.0179 (Represents the average response rate for control group)  
match_ratio1: 0.0029 (Represents the response rate from match ratio 1:1 is 0.29% higher than the control group. Since the p value is 0.097 bigger than 0.05, the result is not statistical significant at the 95% confidence level.)  
match_ratio2: 0.0048 (Represents the response rate from match ratio 2:1 is 0.48% higher than the control group. Since the p value is 0.006 smaller than 0.05, the result is statistical significant at the 95% confidence level.)    
match_ratio3: 0.0049 (Represents the response rate from match ratio 3:1 is 0.49% higher than the control group. Since the p value is 0.005 smaller than 0.05, the result is statistical significant at the 95% confidence level.)  

**Question 3. Calculate the response rate difference between the 1:1 and 2:1 match ratios.**  

Ans:  
Difference betwen the 1:1 and 2:1 match ratios: match_ratio2 - match_ratio1 = 0.0048 - 0.0029 = 0.0019

**Question 4. Based on the regressions you just ran and more generally the results in Table 2a, what do you
conclude regarding the effectiveness of using matched donations?**

Ans:  
From the regression results and table 2a, we can see the positive effect of using matched donations. From regression, match ratio 2:1 increases 0.19% response rate compared to match ratio 1:1 and match ratio 3:1 increases 0.01% response rate compared to match ratio 2:1. Similarly, from table 2a, match ratio 2:1 increases 0.02% response rate compared to match ratio 1:1 and match ratio 3:1 has same response rate with match ratio 2:1. However, we can tell from the number that the increase from match ration 2:1 to 3:1 is smaller than match ratio 1:1 to 2:1. Thus, though the effects are all positive, we should choose match ratio 2:1 to weigh the cost. 

# 3. Response rate in red/ blue states

In [13]:
blue_states = df[df['red_state_dummy'] == 0]
red_states = df[df['red_state_dummy'] == 1]  

reg_blue = smf.ols(formula = 'donation_dummy ~ treatment', data = blue_states)
result_blue = reg_blue.fit()
print(result_blue.summary())
reg_red = smf.ols(formula = 'donation_dummy ~ treatment', data = red_states)
result_red = reg_red.fit()
print(result_red.summary())

                            OLS Regression Results                            
Dep. Variable:         donation_dummy   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                 -0.000
Method:                 Least Squares   F-statistic:                    0.3567
Date:                Sat, 25 Jan 2025   Prob (F-statistic):              0.550
Time:                        13:25:20   Log-Likelihood:                 15783.
No. Observations:               29806   AIC:                        -3.156e+04
Df Residuals:                   29804   BIC:                        -3.155e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.0200      0.001     14.085      0.0

**Question 1. Repeat the regression of response rate on treatment and an intercept (do not include separate match ratio dummies). But this time, base the regression only on respondents in blue states or red states. I.e. run two regressions, one on each of the two sub-samples of data. Interpret the coefficients in both regressions. Is the treatment more effective in red or blue states?**

Ans:  
Blue states:  
Intercept in blue states: 0.02 (When treatment is 0, intercept represents the average response rate for control group)  
Treatment: 0.001 (indicates the treatment group's average response rate is higher than the control group by 0.1%)  
P value: 0.55 is bigger than 0.05 which means the effect is not statistically significant

Red states:  
Intercept in red states: 0.0146 (When treatment is 0, intercept represents the average response rate for control group)  
Treatment: 0.0088 (indicates the treatment group's average response rate is higher than the control group by 0.88%)  
P value: 0.00331 is smaller than 0.05 which means the effect is statistically significant  

According to the data, the treatment in red states is more effective than in blue states (0.88% > 0.1%). 


**Question 2. States are of course not randomly assigned. Does the treatment coefficient have a causal interpretation in each of the two regressions? Does the difference in the treatment effect between states have a causal interpretation?**

Ans:  
Since states are not randomly assigned, difference in culture, politics, demographic exist. Thus, the difference may counfound the treatment and response rate. For instance, people in red states are more willing to donate probably because they more used to charitable atmosphere. This makes it hard to confirm treatment coefficient have a causal effect on response rate. Similarly, we also can't conclude the difference in the treatment effect between two states has a causal effect on response rate. For instance, people in red states are more willing to donate because they live in an abundant life, having extra money for charity. On the other hand, people in blue states can't feed themselves not even to say making donation.

# 4. Response rates and donation amount

In [15]:
reg_money = smf.ols(formula = 'donation_amount ~ treatment', data = df)
result_money = reg_money.fit()
print(result_money.summary())

                            OLS Regression Results                            
Dep. Variable:        donation_amount   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                     3.461
Date:                Sat, 25 Jan 2025   Prob (F-statistic):             0.0628
Time:                        15:03:38   Log-Likelihood:            -1.7946e+05
No. Observations:               50083   AIC:                         3.589e+05
Df Residuals:                   50081   BIC:                         3.589e+05
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.8133      0.067     12.063      0.0

**Question 1. Run a regression of dollars given on a treatment dummy and an intercept. Interpret the regression coefficients. Does the treatment coefficient have a causal interpretation?**

Ans:  
Intercept: 0.8133 (When treatment is 0, intercept represents the average donation amount for control group)
Treatment: 0.1536 (indicates the treatment group's average donation amount is higher than the control group by $ 0.1536)  
P value: 0.0628 is bigger than 0.05 which means the effect is not statistically significant  

We can't conclude the treatment coefficient has a causal effect on donation amount because other than money, other factors like income level, generosity may also affect donation amount.

In [17]:
donation = df[df['donation_dummy'] == 1]
reg_conditional = smf.ols(formula = 'donation_amount ~ treatment', data = donation)
result_conditional = reg_conditional.fit()
print(result_conditional.summary())

                            OLS Regression Results                            
Dep. Variable:        donation_amount   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                 -0.001
Method:                 Least Squares   F-statistic:                    0.3374
Date:                Sat, 25 Jan 2025   Prob (F-statistic):              0.561
Time:                        15:27:32   Log-Likelihood:                -5326.8
No. Observations:                1034   AIC:                         1.066e+04
Df Residuals:                    1032   BIC:                         1.067e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     45.5403      2.423     18.792      0.0

**Question 2. Next, regress dollars given on a treatment dummy and an intercept, but base the regression only on respondents that made a donation (i.e. donation dummy is equal to 1). This regression allows you to analyze how much respondents donate conditional on donating some positive amount. Interpret the regression coefficients. Does the treatment coefficient have a causal interpretation?**

Ans:  
Intercept: 45.5403 (When treatment is 0, intercept represents the average donation amount for control group) Treatment: -1.6684 (indicates the treatment group's average donation amount is less than the control group by $ 0.1536)   
P value: 0.561 is bigger than 0.05 which means the effect is not statistically significant  

The treatment coefficient doesn't have a causal effect on donation money because of selection bias. In this exercise, we only pick people donate money, so the decision to donate is not random. The donation amount can also be affected by people's living condition. For instance, people in the treatment group may be poorer than people in control group, so they donate less money. This kind of factor may confound the result. Thus, the treatment coefficient lacks a causal interpretation.